### ISSUE TITLE
Rules do not detect SnapStart bugs when using dependency injection 

### ISSUE THREAD

**Issue by maschnetwork** [aws/aws-lambda-snapstart-java-rules#14]
Hi,

seems like the plugin does not detect downstream dependencies that are referenced via a DI framework (works when manually creating classes). This is an example that uses Spring Cloud Functions:

**WORKS** when manually initializing the class in the constructor (detects issue):

```java
public class LambdaHandler implements RequestHandler<String, String> {
    private final MyDependency myDependency;

    public LambdaHandler(){
        this.myDependency = new MyDependency();
    }

    @Override
    public String handleRequest(String event, Context context) {
        return myDependency.getUUID().toString();
    }
}

class MyDependency {
    private final UUID random = UUID.randomUUID();

    public UUID getUUID() {
        return random;
    }
}
```

**DOES NOT WORK** when using constructor injection(detects no issues):
```java

@Component
public class LambdaHandler implements RequestHandler<String, String> {
    private final MyDependency myDependency;

    public LambdaHandler(MyDependency myDependency){
        this.myDependency = myDependency;
    }

    @Override
    public String handleRequest(String event, Context context) {
        return myDependency.getUUID().toString();
    }
}

@Component
class MyDependency {
    private final UUID random = UUID.randomUUID();

    public UUID getUUID() {
        return random;
    }
}
```
